src/hooks/useFetch.js

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useState } from "react";
+
+const BASE_URL = `https://api.disneyapi.dev/character`;
+
+export const useFetch = (page = "1", id) => {
+  const [characters, setCharacters] = useState([]);
+  const [info, setInfo] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [errors, setErrors] = useState(null);
+
+  useEffect(() => {
+    const fetchUsers = async () => {
+      setLoading(true);
+      try {
+        const resp = await fetch(`${BASE_URL}/${id ? id : ":id"}?page=${page}`);
+        if (!resp.ok) {
+          throw new Error("Network response was not ok");
+        }
+        const json = await resp.json();
```

Comment:
The name json is misleading because the data it contains is not in that format. A common practice is to use the variable name data.